docs/root/version_history/current.rst

```diff
@@ -27,6 +27,7 @@ Minor Behavior Changes
 * listener: the :ref:`ipv4_compat <envoy_api_field_core.SocketAddress.ipv4_compat>` flag can only be set on Ipv6 address and Ipv4-mapped Ipv6 address. A runtime guard is added ``envoy.reloadable_features.strict_check_on_ipv4_compat`` and the default is true.
 * perf: ssl contexts are now tracked without scan based garbage collection and greatly improved the performance on secret update.
 * router: record upstream request timeouts for all the cases and not just for those requests which are awaiting headers. This behavioral change can be temporarily reverted by setting runtime guard ``envoy.reloadable_features.do_not_await_headers_on_upstream_timeout_to_emit_stats`` to false.
+* runtime: setting deprecated runtime flags to ENVOY_BUG, rather than silently resulting in unexpected behavior on the data plane by no longer applying removed code paths. In the next version of Envoy, Envoy will reject such buggy configuration.
```

Comment:
It is not really BUG, could we have a better name?

Reply:
generally using removed runtime guards is a bug in config, so I think this applies :-)

Reply:
Oh, I see.

Reply:
A couple of things:
1. As currently implemented, I think this just covers static and RTDS layers, which is probably the most important. I think admin overrides won't be caught, but this is probably fine (but maybe call it out?).
2. Would be good to close on the promise of rejecting this configuration.